lib/profile/commands/apply.rb

```diff
@@ -173,6 +177,28 @@ def check_nodes_exist(names=[])
           raise out
         end
       end
+      
+      def expand_brackets(str)
+        labels = []
+        contents = str[/\[.*\]/] 
+        if contents.nil?
+          labels = [str]
+        else
+          left = str[/[^\[]*/]
+          right = str[/].*/][1..-1]
+          if contents.match(/^\[[0-9]+-[0-9]+\]$/)
+            nums = contents[1..-2].split("-")
+            (nums.last.to_i - nums.first.to_i + 1).times do |index|
```

Comment:
You could instead use the `Range` class here, and do:

```ruby
(nums.first..nums.last).each do |i|
  ...
end
```

You would be iterating over each value in the range `start` to `end`. It supports string representations of integers too, which likely simplifies your logic for zero padding.

```ruby

2.7.1 :054 > ("05".."10").each do |i|
2.7.1 :055 >   puts i
2.7.1 :056 > end
05
06
07
08
09
10
 => "05".."10"

2.7.1 :060 > ("0005".."00000010").each do |i|
2.7.1 :061 >   puts i
2.7.1 :062 > end
0005
0006
0007
0008
0009
0010
 => "0005".."00000010"

```

Reply:
Got spot, didn't realise that `Range` would be so cooperative with string representation of integers with leading 0s.